### DISCUSSION TITLE
Nesting directives with indent

### DISCUSSION THREAD

**Discussion by farnabaz** [micromark/micromark#52] [Ideas] [answered]
Adding extra `:` for the parent directive in order to nest another directive inside could be annoying, specially when the content of parent is long.

Is there any possibility to support nest directives with indent, instead of adding extra `:`?

The below code is a sample on how it looks
```
:::parent
Parent content
  :::child
  Child content
  :::
:::
```

**Comment by wooorm** [comment, marked as answer]
How about code and such though?
This project is an implementation of a proposal. You can mention such ideas on the linked thread!

**Reply by farnabaz** [reply]
Thanks for the response, I will write comment on the thread.

Also I wanted to ask, could you give me insights to how it can be achieved using this project?

**Reply by wooorm** [reply]
A lot is possible through extensions (e.g., forking https://github.com/micromark/micromark-extension-directive). But it’s going to be very hard. I would suggest sticking with directives the way they are.

**Reply by ChristianMurphy** [reply]
Another possibility would be using JSX as directives, which are designed to support nesting, and can use indent with nesting https://github.com/micromark/micromark-extension-mdx-jsx
